packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -391,6 +391,10 @@ export abstract class FluidDataStoreContext extends TypedEventEmitter<IFluidData
         const channel = await factory.instantiateDataStore(this, existing);
         assert(channel !== undefined, 0x140 /* "undefined channel on datastore context" */);
         this.bindRuntime(channel);
+
+        // Load the handle to the data store's entryPoint to make sure that the entryPoint initialization function is
+        // called and the entryPoint is fully initialized.
+        await channel.entryPoint?.get();
```

Comment:
thinking longer term i'm not sure how this would all work without request? can someone use entrypoint only? my guess is this is what causes the problem, as it is caused on all realizes. can we find a way to not have this?

Reply:
i think getting rid of this would solve the finishInitialization problem above

Reply:
So the first time the entrypoint is accessed, that would finish initialization, but we would not force that access ourselves when the data store gets realized? I'm thinking of the point you had made previously, of making sure that the whole thing is "constructable" before we put it on the wire and store it... if we don't force that access to the entrypoint during realization, maybe we'd still need that "call" in some code paths right after the realize call?

Starting to see how things behave removing/moving this piece.

Reply:
the call in attach should ensure the thing is constructable before we put in on the wire. would be good to add a test to this affect, specifically, build a data store that cannot be constructed and ensure it cannot be attached